src/common/main.lua

```diff
@@ -1,28 +1,143 @@
 local json = require(".common.json")
 local utils = require(".common.utils")
 local main = {}
+-- just to ignore lint warnings
+local ao = ao or {}
 
 local camel = utils.camelCase
+
+local function errorHandler(err)
+	return debug.traceback(err)
+end
+
 main.init = function()
-	Owner = Owner or ao.env.Process.Owner
-	Name = Name or "ar.io ANT Registry"
+	ANTS = ANTS or {}
+	-- Example ANT structure
+	-- ANTS["antId"] = {
+	--     Owner = "userId",
+	--     Controllers = {"userId1", "userId2"},
```

Comment:
By storing controllers in this way you're not taking advantage of lua's hash-based lookups. It might look silly, but you'll get better lookup performance if you do:
```
{
  Owner = "userId"
  Controllers = {
    ["userId1"]: true,
    ["userId2"]: true
 }
}
```
Then when you want to know if a specific address is in the controllers list, you just do `ant.Controllers[address] ~= nil` and you can just remove a Controller from the table by doing `ant.Controllers[address] = nil`.